src/libsystemd-network/sd-ndisc.c

```diff
@@ -264,52 +265,84 @@ static int ndisc_recv(sd_event_source *s, int fd, uint32_t revents, void *userda
         return ndisc_handle_datagram(nd, rt);
 }
 
+static usec_t ndisc_timeout_compute_random(usec_t val) {
+        return val - val / 10 +
+                (random_u32() % (2 * USEC_PER_SEC)) * val / 10 / USEC_PER_SEC;
```

Comment:
it appears more natural to me to use random_u64() here, given that usec_t is 64bit... Of course, we don't actually need the extra depth, but it doesn't hurt either...